### PR TITLE
Allow configuring RYD API URL

### PR DESCRIPTION
**Pull request by JFronny**
#### What is it?
- [ ] Bugfix (user facing)
- [x] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
This PR introduces an option to configure the API URL for YouTube dislikes.
This is useful for users who whish to use other APIs or proxies such as [RYD-Proxy](https://github.com/TeamPiped/RYD-Proxy) for better privacy.

The UI introduced here is not optimal, as it lacks things like a simple reset-to-default button.
Ideally, the control introduced for configuring the sponsorblock URL would be modified to suit both use cases and reused here.
I am, however, not yet familiar enough with Android application development to attempt to do so.
Should this be a major issue, I'd prefer if someone else took the changes here and implemented this improvement by themselves.

#### APK testing

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/res/values-de/strings.xml [hunk 1]
```diff
@@ -718,4 +718,6 @@
     <string name="streams_not_yet_supported_removed">Streams, die noch nicht vom Downloader unterstützt werden, werden nicht angezeigt</string>
     <string name="selected_stream_external_player_not_supported">Der ausgewählte Stream wird von externen Playern nicht unterstützt</string>
     <string name="progressive_load_interval_title">Größe des Ladeintervalls für die Wiedergabe</string>
+    <string name="return_youtube_dislike_api_url_summary">Der URL von dem Dislikes abgerufen werden sollen.</string>
```
Comment (Atemu) [inline app/src/main/res/values-de/strings.xml:721]:
```suggestion
    <string name="return_youtube_dislike_api_url_summary">Die URL von der Dislikes abgerufen werden sollen.</string>
```

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -821,4 +821,7 @@
     <string name="disable_error_reports_summary">Prevent all error reporting screens from appearing. This may result in the app behaving unexpectedly. <b>USE AT YOUR OWN RISK!</b></string>
     <string name="enable_return_youtube_dislike_title">Show Dislike Count</string>
     <string name="enable_return_youtube_dislike_summary">Use the ReturnYouTubeDislike API to show the amount of dislikes for a video. This only works for YouTube videos.\n<b>WARNING: Your IP address will be visible to the API. Use at your own risk!</b></string>
+    <string name="return_youtube_dislike_default_api_url">https://returnyoutubedislikeapi.com/votes?videoId=</string>
+    <string name="return_youtube_dislike_api_url_summary">The url to use when querying dislikes.</string>
```
Comment (Atemu) [inline app/src/main/res/values/strings.xml:825]:
```suggestion
    <string name="return_youtube_dislike_api_url_summary">URL to query dislikes from.</string>
```

**Comment by polymorphicshade** [comment]
Thank you for this - I've already made changes related to this in Tubular.

**Comment by JFronny** [comment]
What is `Tubular`? The feature seems to still be missing here.

**Comment by polymorphicshade** [comment]
> What is `Tubular`? The feature seems to still be missing here.

https://github.com/polymorphicshade/NewPipe/discussions/164

**Pull request closed without merging**